README.md

```diff
@@ -110,17 +113,47 @@ It's trivial!
 $ nix-shell ~/ghc.nix/shell.nix --arg nixpkgs '(import <nixpkgs> {}).pkgsi686Linux'
 ```
 
-## Building a WebAsm cross-compiler
+## Building a WebAsm or JavaScript cross-compiler
+
+Both cross-compilers are supported with `nix-shell` or the flake-based `nix develop`.
+
+`CC`, `CONFIGURE_ARGS`, etc. environment variables will be overridden to configure the cross-compiler.
+
+HLS should also just work.
+
+### For WebAsm:
+
+```sh
+nix-shell ~/ghc.nix --arg withWasiSDK true
+# or
+nix develop github:alpmestan/ghc.nix#wasi-cross
+```
+
+Once in the shell,
+
+```sh
+$ ./boot
+$ configure_ghc
+$ hadrian/build --docs=none
+# or for a faster experience,
+$ hadrian/build -j4 --flavour=quick+no_profiled_libs --docs=none
+```
+
+### For JavaScript:
+
+```sh
+nix-shell ~/ghc.nix --arg withEMSDK true
+# or
+nix develop github:alpmestan/ghc.nix#js-cross
+```
```

Comment:
I feel like we should only add the arguments to the example commands that are really needed, e.g. there's no motivation to add `-j4` here.

Reply:
I'm perfectly torn on the matter.

Actually, I think I can just remove the example use of hadrian altogether, because it's redundant. Huzzah!